indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/TaskLockConfig.java

```diff
@@ -36,6 +36,9 @@ public class TaskLockConfig
   @JsonProperty
   private long batchAllocationWaitTime = 0L;
 
+  @JsonProperty
+  private boolean segmentAllocationReduceMetadataIO = false;
```

Comment:
This is used only for batch segment allocation, IIUC. Let's rename it to `batchAllocationReduceMetadataIO`.
Once this has been tested thoroughly, we can remove the flag altogether and use the new approach for both regular and batch allocation.

Reply:
Are you against its usage in the normal flow with the same feature flag?
It's behind a flag anyway so I don't think there's a need to not make the changes there.